mistletoe/block_token.py

```diff
@@ -1058,7 +1064,7 @@ def read(cls, lines):
                 if cls._end_cond in line.casefold():
                     break
             elif line.strip() == '':
-                line_buffer.pop()
+                lines.backstep()
```

Comment:
It doesn't really matter for the HtmlBlock, because the trailing endlines are stripped off in `__init__` anyway. But I'll add it back for consistency.